modules/standard/Socket.md

```diff
@@ -0,0 +1,210 @@
+- udpSocket : A connection less UDP Socket
+
+## Address Information
+
+`ipAddr` serves the purpose of storing address information in a single record irrespective of the protocol used which varies the buffer required to store address.
+
+To initialize an `ipAddr` there are several available overloads and defaults.
+
+**Method**
+```python
+proc ipAddr.init(host:string = "localhost", port:uint(16) = 8000, family:IPFamily = IPFamily.IPv4)
+```
+**Parameters**
+```
+host: string - address string in dot or color notation depending on family. "127.0.0.1" or "::1"
+port: uint(16) - port number for socket
+family: IPFamily - Internet Protocol version which can be IPFamily.IPv4 or IPFamily.IPv6
+```
+
+There are two more initialzers which can take in standard IP Addresses like loopback and broadcast as `host` which are defined as constants inside the `Sys` module.
+
+**Method**
+```python
+proc ipAddr.init(host:sys_in_addr_t, port: uint(16) = 8000)
+proc ipAddr.init(host:sys_in6_addr_t, port: uint(16) = 8000)
+```
+**Parameter**
+```
+host: string - standard ip addresses
+port: uint(16) - port number for socket
+```
+
+Family is determined based on type of standard address used `sys_in_addr_t` for ipv4 and `sys_in6_addr_t` for ipv6. Details about available options can be found [here](https://www.ibm.com/docs/en/zos/2.3.0?topic=considerations-special-ip-addresses#ipv6d0131000420__specialip)
+
+
+`ipAddr` instance created will then have getters for all the three arguments i.e. `host`, `port` and `family` which dont take up any space inside the record and are fetched from [sockaddr_storage](https://www.usenix.org/legacy/publications/library/proceedings/usenix2000/freenix/full_papers/metzprotocol/metzprotocol_html/node10.html)  using [getnameinfo](https://www.ibm.com/docs/en/zos/2.4.0?topic=reference-getnameinfo-flags-returned-information-examples#getnameinfofe) internally. [sockaddr_storage] is the only thing `ipAddr` stores and takes up memory for.
+
+## Connecting to remote sockets
+
+`connect` function will create `tcpConn` instance which will take in the following parameters
+
+**Method:**
+```python
+proc connect(addr:ipAddr) : tcpConn
+```
+
+**Parameters:**
+
+```
+addr: ipAddr - address information of remote socket
+```
+
+A higher level construct for socket will take in `host` as `string` which can perform DNS resolution and convert to `ipAddr` internally.
+
+**Method:**
+```python
+proc connect(host:string, port:int, family:int) : tcpConn
+```
+
+**Parameters:**
+```
+host: String - host name either in dot-dash or domain form default "localhost"
+port: Integer - port number
+family: Integer - AF_INET or AF_INET6 or AF_UNSPEC
+```
+
+The purpose of family is important for address resolution and converting addr string to standard notation as well as validation. How the family can affect address resolution depends on [getaddrinfo](https://www.ibm.com/docs/en/zos/2.1.0?topic=functions-getaddrinfo-get-address-information#getaddrinfo) function.
```

Comment:
To the extent we can, we should reference primary sources from standards bodies rather than secondary sources from vendors or implementors. They're both more authoritative and (I would guess) more likely to stay available at the same URL into the future. So for this one, how about `https://pubs.opengroup.org/onlinepubs/9699919799/functions/getaddrinfo.html` (POSIX) instead of the IBM doc?
(Looking at the others I don't know that we'll be able to do this for all of them, but I'm going to try.)

Reply:
certainly, my idea was although to provide the reader a place from where they can read about it with ease never thought about future availability but primary sources should be used instead so will make changes :)